File: internal/storage/postgres/mapper/product.go
package mapper

import (
	model "hardware_store/internal/model/product"
	"hardware_store/internal/storage/postgres/dto"
)

func ProductToDTO(p model.Product) dto.ProductDTO {
	return dto.ProductDTO{
		ProductID:      p.ProductID,
		Name:           p.Name,
		CategoryID:     p.CategoryID,
		Price:          p.Price,
		AvailableStock: p.AvailableStock,
		LastUpdateDate: p.LastUpdateDate,
		SupplierID:     p.SupplierID,
		ImageID:        p.ImageID,
	}
}

func ProductFromDTO(d dto.ProductDTO) model.Product {
	return model.Product{
		ProductID:      d.ProductID,
		Name:           d.Name,
		CategoryID:     d.CategoryID,
		Price:          d.Price,
		AvailableStock: d.AvailableStock,
		LastUpdateDate: d.LastUpdateDate,
		SupplierID:     d.SupplierID,
		ImageID:        d.ImageID,
	}
}

File: store_python/src/app/models/product.py
import uuid
from datetime import date, datetime
from decimal import Decimal
from app.extensions import db


class Product(db.Model):
    """DAO: таблица товаров."""
    __tablename__ = "product"

    id = db.Column(db.Uuid(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = db.Column(db.String(200), nullable=False)
    category = db.Column(db.String(100), nullable=False)
    price = db.Column(db.Numeric(12, 2), nullable=False)
    available_stock = db.Column(db.Integer, nullable=False, default=0)
    last_update_date = db.Column(db.Date, nullable=True)
    supplier_id = db.Column(db.Uuid(as_uuid=True), db.ForeignKey("supplier.id"), nullable=False)
    # Image bytes are stored in external photo-service; we keep only image UUID here.
    image_id = db.Column(db.Uuid(as_uuid=True), nullable=True)

    supplier = db.relationship("Supplier", backref="products", foreign_keys=[supplier_id])
